hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/dynamicconfig/AddCacheConfigMessageTask.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.impl.protocol.task.dynamicconfig;
+
+import com.hazelcast.client.impl.protocol.ClientMessage;
+import com.hazelcast.client.impl.protocol.codec.DynamicConfigAddCacheConfigCodec;
+import com.hazelcast.config.CachePartitionLostListenerConfig;
+import com.hazelcast.config.CacheSimpleConfig;
+import com.hazelcast.config.CacheSimpleConfig.ExpiryPolicyFactoryConfig;
+import com.hazelcast.config.InMemoryFormat;
+import com.hazelcast.instance.Node;
+import com.hazelcast.internal.dynamicconfig.AddDynamicConfigOperationFactory;
+import com.hazelcast.nio.Connection;
+import com.hazelcast.spi.OperationFactory;
+
+import java.util.List;
+
+public class AddCacheConfigMessageTask
+        extends AbstractAddConfigMessageTask<DynamicConfigAddCacheConfigCodec.RequestParameters> {
+
+    public AddCacheConfigMessageTask(ClientMessage clientMessage, Node node, Connection connection) {
+        super(clientMessage, node, connection);
+    }
+
+    @Override
+    protected DynamicConfigAddCacheConfigCodec.RequestParameters decodeClientMessage(ClientMessage clientMessage) {
+        return DynamicConfigAddCacheConfigCodec.decodeRequest(clientMessage);
+    }
+
+    @Override
+    protected ClientMessage encodeResponse(Object response) {
+        return DynamicConfigAddCacheConfigCodec.encodeResponse();
+    }
+
+    @Override
+    protected OperationFactory getOperationFactory() {
+        CacheSimpleConfig config = new CacheSimpleConfig();
+        config.setAsyncBackupCount(parameters.asyncBackupCount);
```

Comment:
Hmm, yes, we're not cloning here, we are constructing. Stupid me. I'm just trying to find a way to prevent all this scattered code duplication, which will make it very hard to refactor anything regarding config in the future.

Are not most (or all) affected config classes `IdentifiedDataSerializable`, so we can let the codec deserialize a instance? `CacheSimpleConfig` is not IDS, but can we make it IDS without breaking RU?